File: app/src/main/java/com/wealoha/social/beans/Profeature.java
package com.wealoha.social.beans;

import android.content.Context;

public class Profeature {

	private String title;
	private String itemTitle;
	private String itemContent;
	private int headerResId;

	public Profeature(String title, String itemTitle, String itemContent, int headerResId) {
		super();
		this.title = title;
		this.itemTitle = itemTitle;
		this.itemContent = itemContent;
		this.headerResId = headerResId;
	}

	public Profeature(Context context, int itemTitleId, int itemContentId, int stateid, int headerResId) {
		title = getString(context, itemTitleId);
		itemTitle = getString(context, itemContentId);
		itemContent = getString(context, stateid);
		this.headerResId = headerResId;
	}

	private String getString(Context context, int stringId) {
		if (stringId == 0) {
			return null;
		} else {
			return context.getResources().getString(stringId);
		}
	}

	public String getTitle() {
		return title;
	}

	public String getExplain() {
		return itemTitle;
	}

	public String getState() {
		return itemContent;
	}

	public int getHeaderResId() {
		return headerResId;
	}

}

File: app/src/main/java/com/wealoha/social/beans/Profeature.kt
package com.wealoha.social.beans

import android.content.Context

class Profeature {
    var title: String?
        private set
    var explain: String?
        private set
    var state: String?
        private set
    var headerResId: Int
        private set

    constructor(
        title: String?,
        itemTitle: String?,
        itemContent: String?,
        headerResId: Int
    ) : super() {
        this.title = title
        explain = itemTitle
        state = itemContent
        this.headerResId = headerResId
    }

    constructor(
        context: Context,
        itemTitleId: Int,
        itemContentId: Int,
        stateid: Int,
        headerResId: Int
    ) {
        title = getString(context, itemTitleId)
        explain = getString(context, itemContentId)
        state = getString(context, stateid)
        this.headerResId = headerResId
    }

    private fun getString(context: Context, stringId: Int): String? {
        return if (stringId == 0) {
            null
        } else {
            context.resources.getString(stringId)
        }
    }

}